utils/zerovec/src/map/vecs.rs

```diff
@@ -35,6 +37,19 @@ pub trait ZeroVecLike<'a, T: ?Sized> {
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    /// Convert T to a needle for searching
+    fn t_as_needle(t: &T) -> &T;
+
+    /// Compare this type with a `Self::GetType`. This must produce the same result as
+    /// if `g` were converted to `Self`
+    fn t_cmp_get(t: &T, g: &Self::GetType) -> Ordering;
```

Comment:
Thought/Suggestion: Should we bind `Self::GetType` to `PartialOrd<T>` instead of adding this method?

Reply:
We could but this puts more requirements on the individual types, whereas in the current model people who wish to use a type in ZeroMap only need to implement ZeroMapKV which involves setting a couple associated types.